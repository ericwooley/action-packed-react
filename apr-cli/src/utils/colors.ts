import chalk from "chalk";

export const { green, grey, blue, red } = chalk;
