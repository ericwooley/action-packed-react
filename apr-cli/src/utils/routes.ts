import debug from "debug";
import glob from "glob";
import { join, sep } from "path";
import fs from "fs";

export function findRoutes(name: string, { pretty = false } = {}) {
  const log = debug("apr:generate:" + name);
  const dir = join(process.cwd(), "/**/routes/*");

  log("scanning: ", dir);
  let routes = glob.sync(dir);
  log("unprocessed routes", routes, sep);
  routes = routes
    .filter(file => fs.statSync(file).isDirectory())
    .filter(file => file !== "." && file !== "..")
    .map(route => {
      const formatted = route
        .split(sep)

        .join(sep)
        .replace(`${process.cwd()}${sep}`, "");
      if (pretty) return formatted.replace(new RegExp(`${sep}routes${sep}`, "g"), " -> ");
      return formatted;
    });
  log("routes: ", routes);
  return ["src/", ...routes];
}

const routeStr = `${sep}routes${sep}`;
const prettySep = " -> ";
const routeSepRGX = new RegExp(routeStr, "g");

export function prettyRoute(route: string, {newSep = prettySep} = {}) {
  return route.replace(routeSepRGX, newSep).replace(/^src\//, '/');
}
export function uglyRoute(route: string, {oldSep = prettySep} = {}) {
  return route.replace(new RegExp(oldSep, "g"), routeStr).replace(/^\//, 'src\//');
}
