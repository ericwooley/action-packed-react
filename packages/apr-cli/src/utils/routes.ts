import debug from "debug";
import glob from "glob";
import { join, sep } from "path";
import fs from "fs";
const routeLog = debug("apr:foundRoutes")
export function findRoutes(name: string, { pretty = false } = {}) {
  const log = debug("apr:generate:" + name);
  const dir = join(process.cwd(), "./**/routes");

  log("scanning: ", dir);
  let routes = glob.sync(dir);
  log("unprocessed routes", routes, sep);
  routes = routes
    .filter(file => fs.statSync(file).isDirectory())
    .filter(file => file !== "." && file !== "..")
    .map(route => {
      const formatted = route
      .replace(new RegExp(`${sep}routes$`), '')
      .split(sep)
      .join(sep)
      .replace(`${process.cwd()}${sep}`, "")
      // glob bug????
        .replace(/^srcroutes/, `src${sep}routes`);
      if (pretty) return prettyRoute(formatted);
      return formatted;
    });
  log("routes: ", routes);
  return [...routes];
}

const routeStr = `${sep}routes${sep}`;
const routeSepRGX = new RegExp(routeStr, "g");
const prettySep = " -> ";
routeLog("routeStr", routeStr)

export function prettyRoute(route: string, { newSep = prettySep } = {}) {
  const pretty = route.replace(routeSepRGX, newSep);
  routeLog("pretty -", route, ': |', pretty, '| sep=', prettySep)
  return pretty
}
export function uglyRoute(route: string, { oldSep = prettySep } = {}) {
  const ugly = route.replace(new RegExp(oldSep, "g"), routeStr);
  routeLog(route, ': |', ugly, '| sep=', prettySep)
  return ugly
}
