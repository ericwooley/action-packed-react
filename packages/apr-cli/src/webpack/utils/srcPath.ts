const path = require('path')
module.exports = function srcPath (subdir: string) {
  const result = path.join(process.cwd(), 'src', subdir)
  return result
}
