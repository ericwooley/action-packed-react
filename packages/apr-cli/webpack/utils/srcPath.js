const path = require('path')
module.exports = function srcPath (subdir) {
  const result = path.join(process.cwd(), 'src', subdir)
  return result
}
