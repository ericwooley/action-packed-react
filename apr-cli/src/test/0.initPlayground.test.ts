import { spawnSync } from "child_process";
import { snapshotPlayground, PlaygroundWatcher } from "./utils/getAllSourceFiles";

beforeAll(() => {
  process.chdir("../");
  spawnSync("yarn", ["playground-init", "--dev"], { stdio: "inherit" });
});
describe("setup", () => {
  let watcher: PlaygroundWatcher;
  beforeEach(() => {
    watcher = new PlaygroundWatcher();
  });
  afterEach(async () => {
    await watcher.stop();
  });
  it("should create the initial files", async () => {
    await snapshotPlayground(watcher);
  });
});
