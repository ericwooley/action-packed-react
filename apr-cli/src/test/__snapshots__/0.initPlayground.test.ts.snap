// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setup should create the initial files: <project-root>playground/apr.config.js 1`] = `
"module.exports = {
  /* Webpack dev server options */
  // port: 8080,                         // Webpack dev server port
  // host: \\"localhost\\",                  // Webpack dev server host
}
"
`;

exports[`setup should create the initial files: <project-root>playground/package.json 1`] = `
"{
  \\"name\\": \\"playground\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"My action-packed-react app\\",
  \\"main\\": \\"webpack.config.js\\",
  \\"scripts\\": {
    \\"prebuild\\": \\"rm -rf dist\\",
    \\"build\\": \\"apr build\\",
    \\"start\\": \\"apr dev\\",
    \\"lint\\": \\"tslint --project tsconfig.json\\",
    \\"test\\": \\"jest --passWithNoTests\\"
  },
  \\"author\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"dependencies\\": {
    \\"action-packed-react\\": \\"<project-root>action-packed-react\\",
    \\"apr-cli\\": \\"<project-root>apr-cli\\",
    \\"history\\": \\"^4.10.1\\",
    \\"react\\": \\"^16.12.0\\",
    \\"react-dom\\": \\"^16.12.0\\",
    \\"styled-components\\": \\"^4.4.1\\"
  },
  \\"devDependencies\\": {
    \\"@storybook/addon-info\\": \\"^5.2.8\\",
    \\"@storybook/addon-storyshots\\": \\"^5.2.8\\",
    \\"@storybook/react\\": \\"^5.2.8\\",
    \\"@types/history\\": \\"^4.7.3\\",
    \\"@types/react\\": \\"^16.9.15\\",
    \\"@types/react-dom\\": \\"^16.9.4\\",
    \\"@types/styled-components\\": \\"^4.4.0\\",
    \\"babel-jest\\": \\"^24.9.0\\",
    \\"jest\\": \\"^24.9.0\\"
  }
}
"
`;

exports[`setup should create the initial files: <project-root>playground/tsconfig.json 1`] = `
"{
  \\"compilerOptions\\": {
    \\"target\\": \\"es5\\",
    \\"lib\\": [\\"es2015\\", \\"es2016\\", \\"es2017\\", \\"dom\\"],
    \\"strict\\": true,
    \\"allowSyntheticDefaultImports\\": true,
    \\"experimentalDecorators\\": true,
    \\"emitDecoratorMetadata\\": true,
    \\"jsx\\": \\"react\\",
    \\"typeRoots\\": [\\"node_modules/@types\\", \\"src/types\\"],
    \\"rootDirs\\": [\\"src\\"],
    \\"baseUrl\\": \\"src\\",
    \\"paths\\": {
      \\"app\\": [\\"index\\"],
      \\"app/*\\": [\\"*\\"]
    }
  },
  \\"exclude\\": [\\"**/*.test.ts*\\", \\"\\"]
}
"
`;

exports[`setup should create the initial files: file-list 1`] = `
Array [
  "<project-root>playground/src/components/RootLayout/index.tsx",
  "<project-root>playground/src/index.tsx",
  "<project-root>playground/src/redux/ducks/index.ts",
  "<project-root>playground/src/redux/sagas/index.ts",
  "<project-root>playground/src/redux/selectors/index.ts",
  "<project-root>playground/src/route.story.tsx",
  "<project-root>playground/src/route.tsx",
  "<project-root>playground/src/routes/readme.md",
  "<project-root>playground/src/ui/readme.md",
]
`;

exports[`setup should create the initial files: files-changed 1`] = `
Array [
  "<project-root>playground/apr.config.js",
  "<project-root>playground/package.json",
  "<project-root>playground/tsconfig.json",
]
`;
